docs/attestation.md

```diff
@@ -0,0 +1,120 @@
+# Set up remote attestation
+
+## Android side
+
+ 1. Before you can enable the Auditor app in your configuration you have to
+    generate a signing key.  There is currently no script generated for this, but
+    it's easy enough to do using the `generateKeysScript` target.
+    ```console
+    $ nix-build \
+    	--arg configuration ./sunfish.nix \
+    	-A generateKeysScript \
+    	-o generate-keys
+    $ cp -L generate-keys auditor-keys
+    ```
+    Then delete the line with the assignment of the `KEYS`variable and replace it
+    with `KEYS=( auditor )`.  I have also changed the common name of the
+    certificate to `Robotnix auditor` because it is not device dependent.  The
+    resulting script should look something like this (with potentially different
+    hashes of course):
+    ```bash
+    #!/nix/store/2jysm3dfsgby5sw5jgj43qjrb5v79ms9-bash-4.4-p23/bin/bash
+    set -euo pipefail
+
+    export PATH=/nix/store/q0ajpzppqfrlbzbddbbzv1w6vfzydhk5-openssl-1.1.1g-bin/bin:/nix/store/8plhh65p17qlyp7k74vaiisyrhg15hwr-android-key-tools/bin:$PATH
+
+    KEYS=( auditor )
+
+    for key in "${KEYS[@]}"; do
+      if [[ ! -e "$key".pk8 ]]; then
+        echo "Generating $key key"
+        # make_key exits with unsuccessful code 1 instead of 0
+        make_key "$key" "/CN=Robotnix auditor/" && exit 1
+      else
+        echo "Skipping generating $key since it is already exists"
+      fi
+    done
+    ```
+    Run the script to generate the keys:
+    ```bash
+    $ cd keys/
+    $ bash ../auditor-keys
+    ```
+
+ 2. Now you can enable the Auditor app in the configuration:
+    ```nix
+    {
+      keyStorePath = "/dev/shm/android-keys";
+      signing.enable = true;
+
+      apps = {
+        auditor.enable = true;
+        auditor.domain = "attestation.example.com";
+      };
+    }
+    ```
+    You also need to have signing enabled during build time.  Trying to build
+    for later signing with the release script doesn't build. (TODO: Probably a bug)
```

Comment:
This actually isn't a bug.  The auditor app needs to know it's own signing key fingerprint since it includes it in its source. So, we have to have at least set `keyStorePath` that has an `auditor.x509.pem` file that it can read.  We should probably set an assertion / warning in the auditor module to enforce this.